routing/router.go

```diff
@@ -1402,6 +1402,10 @@ func (r *ChannelRouter) FindRoute(source, target route.Vertex,
 		finalCLTVDelta = finalExpiry[0]
 	}
 
+	// Add BlockPadding to the finalCltvDelta so that the receiving node
+	// does not reject the HTLC if a block is mined while its in-flight.
+	finalCLTVDelta += BlockPadding
```

Comment:
I am in doubt whether we should also pad when the user specified an explicit final cltv expiry for `queryroutes`. It is a more low level function and it may be confusing if it doesn't returns what was requested?

Maybe the same thing holds for `sendpayment` with a set expiry? Perhaps the padding should just happen when only an encoded payment request is supplied.

Will get back to you.

Reply:
@Crypt-iQ it seems best to only apply the padding if no explicit final cltv delta is set in the rpc request.

Reply:
looks like this still needs to be moved

Reply:
Will do this week

Reply:
Addressed the comments.  This is related to #3150 since if there is no expiry set, the payment request will likely fail because of sender-side checks (at least if the node is an lnd node).  The result of that discussion was to *require* the final cltv delta.  Thoughts?